### PR TITLE
Add tests on cube download routes

### PR DESCRIPTION
**Pull request by uponthesun**
Uses [mongodb-memory-server](https://www.npmjs.com/package/mongodb-memory-server) for an in-memory DB for the tests, and [supertest](https://www.npmjs.com/package/supertest) to make requests and verify the response.

I did one test on each download format with some basic expectations. It seems like it'd be more difficult to do more comprehensive testing on all the branches without more cube fixture data, which I don't know how to generate efficiently.

Also added one test in another file just to ensure this approach is generalizable and there isn't weirdness with multiple tests using mongodb-memory-server.

### CODE REVIEW

File: __tests__/routes/cube/download.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  await dbSetup.close(mongoServer);
+  carddb.unloadCardDb();
+});
+
+test('text download', () => {
+  return request(app)
+    .get('/cubecobra/' + cubeID)
+    .expect('Content-Type', 'text/plain')
+    .expect(200)
+    .expect(function (res) {
```
Comment (ruler501) [inline __tests__/routes/cube/download.test.js:36]: Should probably make this a lambda to match everything else.

**Review by ruler501 — CHANGES_REQUESTED** [review]
Looks like a great start. Very happy to see testing getting worked on.

**Review by joshfinnie — DISMISSED** [review]
excellent!

**Review by ruler501 — CHANGES_REQUESTED** [review]
I'm not sure that I like the strategy of checking string equality in general here. I'm not sure there's a better general framework without a way to parse the different formats.
Handling adventures, split cards, MDFCs, and preflipped TDFCs would be good edge cases to handle, but can be brought up in a later pr.

File: __tests__/routes/cube/api.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+const fixturesPath = 'fixtures';
+const exampleCube = cubefixture.exampleCube;
+const cubeID = exampleCube.shortID;
+
+app.use('/', router);
+
+let mongoServer;
+
+beforeAll(async () => {
```
[the remark this answers is not in the document]
Comment (uponthesun) [reply __tests__/routes/cube/api.test.js:20]: I'm not super familiar with JS async stuff, but I think so:

https://jestjs.io/docs/setup-teardown#one-time-setup
https://jestjs.io/docs/asynchronous#promises

So if the function returns a promise, it'll wait for the promise to resolve before proceeding. And I think an async function always returns a promise.

The tests also pass consistently (and fail if the beforeAll is removed), if that helps.

File: __tests__/routes/cube/download.test.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      expect(lines[0]).toEqual('1 Acclaimed Contender');
+      expect(lines[1]).toEqual('2 Brazen Borrower');
+      // The two Brazen Borrowers in the cube are deduped
+      expect(lines.length).toEqual(exampleCube.cards.length - 1);
+    });
+});
+
+test('csv download', () => {
+  const headerLine =
+    'Name,CMC,Type,Color,Set,Collector Number,Rarity,Color Category,Status,Finish,Maybeboard,Image URL,Image Back URL,Tags,Notes,MTGO ID';
```
[the remark this answers is not in the document]
Comment (uponthesun) [reply __tests__/routes/cube/download.test.js:74]: I think the additional validations are a good idea and I'll look into them, but is the "fragility" here actually a bad thing? I see it as the most straightforward way to test the contract of the CSV download API. Any changes to the columns here is a change to that contract and I don't think those should happen willy-nilly. Do those changes happen more frequently than I'm imagining?

[the remark this answers is not in the document]
Comment (uponthesun) [reply __tests__/routes/cube/download.test.js:83]: I added an assertion that the output parses correctly as a CSV. In the process I discovered that the current code actually includes an extra comma at the end of the line, which papaparse can still accept since it's tolerant, but it does generate an error. I went ahead and removed that extra comma from writeCard, which I think is safe since it doesn't seem to be used anywhere else, but LMK if we need to keep it as-is for some reason.

IMO this set of assertions - valid CSV format, header line and first line are equal to hardcoded expectation - is where I want this test to be, to balance test simplicity with comprehensiveness.

There should also be a "round-trip" test for each format that ensures that downloading and then importing works correctly, but these will be a little more involved so I'd like to not tackle them in this PR, if that's OK.

[the remark this answers is not in the document]
Comment (ruler501) [reply __tests__/routes/cube/download.test.js:83]: The contract for our csv is very specifically not dependent on the order of columns and accepting of adding and removing(to an extent) columns.

[the remark this answers is not in the document]
Comment (uponthesun) [reply __tests__/routes/cube/download.test.js:83]: I buy that it's not dependent on the order and I can update the test accordingly, but it seems like it's still fair to have a test update when you add or remove a column. Is that ok?

As someone who consumes your API for a different use case than re-importing into Cubecobra, I'd like to think there's some degree of stability on the data format.

**Comment by ruler501** [comment]
That seems reasonable then.

**Review by ruler501 — APPROVED** [review]
